### PR TITLE
release: fix release regex

### PR DESCRIPTION
**Pull request by rail**
In #95332 we changed the way how a release branch is detected. Prior to that change, if a build was not a custom one, we assumed the branch was a release branch.

With addition of customized build logic, we changed the release branch detection logic to be explicit, but we missed a few exceptions.

This PR adds `staging` and `rc` prefixes to the release branch detection logic.

Epic: none
Release note: None

### CODE REVIEW

**Review by rickystewart — APPROVED** [review]

**Comment by rail** [comment]
bors r=rickystewart

**Comment by rail** [comment]
TYFTR!

File: build/teamcity/internal/release/process/make-and-publish-build-artifacts.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ build_name=$(git describe --tags --dirty --match=v[0-9]* 2> /dev/null || git rev
 # On no match, `grep -Eo` returns 1. `|| echo""` makes the script not error.
 release_branch="$(echo "$build_name" | grep -Eo "^v[0-9]+\.[0-9]+" || echo"")"
 is_customized_build="$(echo "$TC_BUILD_BRANCH" | grep -Eo "^custombuild-" || echo "")"
-is_release_build="$(echo "$TC_BUILD_BRANCH" | grep -Eo "^(release-[0-9][0-9]\.[0-9](\.0)?)$|master$" || echo "")"
+is_release_build="$(echo "$TC_BUILD_BRANCH" | grep -Eo "^((staging|release|rc)-(v)?[0-9][0-9]\.[0-9](\.0)?).*|master$" || echo "")"
```
Comment (celiala) [inline build/teamcity/internal/release/process/make-and-publish-build-artifacts.sh:16]: good thinking to also add the rc- prefix!

**Review by celiala — APPROVED** [review]
thank you!

**Comment by rail** [comment]
bors r=celiala

**Pull request merged**